### PR TITLE
Docs: Add TS definition import example for API doc

### PR DESCRIPTION
**Pull request by RyannosaurusRex**
## Pull request checklist

Make sure you:

- [X] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [X] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [X] Added/Updated related documentation.
- [X] Added/Updated related tests.

## Short description of the change(s)

Adds an example of how to import the TypeScript type definitions needed for the example to successfully build.

### CODE REVIEW

File: packages/hint/docs/user-guide/api/using-api.md [hunk 1]
```diff
@@ -14,6 +14,15 @@ To use the API, the first thing you need to do is import the class `Analyzer`.
 import { Analyzer } from 'hint';
 ```
 
+This example is using TypeScript, so you'll also need to import some type definitions to
+enable Intellisense support and a successful TypeScript build. Any types you need are included with Webhint and are
+located in the `hint/dist/src/lib/types` directory. Two types we'll be using in the example are
```
Comment (molant) [inline packages/hint/docs/user-guide/api/using-api.md]: Hi @RyannosaurusRex 👋

Nice catch! One question I have is about the import path. We are [exporting the types directly on hint](https://github.com/webhintio/hint/blob/314bc896711b361f6cdf28f5b95ff4334fb99a92/packages/hint/src/lib/index.ts#L3) so people shouldn't have to access `hint/dist` (`dist` should be considered private). Can you see if that works in your code and update the message accordingly?

Thanks!

File: packages/hint/docs/user-guide/api/using-api.md [hunk 2]
```diff
@@ -14,6 +14,15 @@ To use the API, the first thing you need to do is import the class `Analyzer`.
...
+`UserConfig` and `AnalyzerResult`, so import those with the following line:
+
+ ```ts
+import { UserConfig, AnalyzerResult } from 'hint/dist/src/lib/types';
```
Comment (molant) [inline packages/hint/docs/user-guide/api/using-api.md]:
```suggestion
import { UserConfig, AnalyzerResult } from 'hint';
```
Based on my previous comment. If this doesn't work then we should fix something on our side.

**Comment by RyannosaurusRex** [comment]
@molant I thought I got an error trying that before, but I just changed it back and works now. 🤷‍♂ 😄 

Your line suggestion works for me. 👍

**Comment by molant** [comment]
@RyannosaurusRex merged 🎉

Should be in the site in a few minutes 🤞

**Pull request closed without merging**